next/src/components/common/NavBar/NavMenu/NavMenuContextProvider.tsx

```
@@ -0,0 +1,36 @@
+import React, { createContext, PropsWithChildren, useContext, useState } from 'react'
+
+type NavMenuContextType = {
+  menuValue: string
+  setMenuValue: (value: string) => void
+  isMobileMenuOpen: boolean
+  setMobileMenuOpen: (value: boolean) => void
+}
+
+const NavMenuContext = createContext<NavMenuContextType>({
+  menuValue: '',
+  setMenuValue: () => {},
+  isMobileMenuOpen: false,
+  setMobileMenuOpen: () => {},
+})
+
+/**
+ * Based on bratislava.sk: https://github.com/bratislava/bratislava.sk/blob/master/next/components/common/NavBar/NavMenu/navMenuContext.tsx
+ */
+
+export const NavMenuContextProvider = ({ children }: PropsWithChildren<{}>) => {
+  const [menuValue, setMenuValue] = useState('')
+  const [isMobileMenuOpen, setMobileMenuOpen] = useState(false)
+
+  return (
+    <NavMenuContext.Provider
+      value={{ menuValue, setMenuValue, isMobileMenuOpen, setMobileMenuOpen }}
+    >
+      {children}
+    </NavMenuContext.Provider>
+  )
+}
+
+export const useNavMenuContext = () => {
+  return useContext(NavMenuContext)
+}
```

Please also throw an error, when used outside of the provider, similar to `GeneralContextProvidex.tsx`. In this case:
```tsx
export const useNavMenuContext = () => {
  const context = useContext(GeneralContext)

  if (!context) {
    throw new Error('The useNavMenuContext hook must be used within a NavMenuContextProvider')
  }

  return context
}
```